Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1353,87 +1353,118 @@ static LayoutUnit computeSubgridMarginBorderPadding(const RenderGrid* outermost,
     return subgridMbp;
 }
 
-void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track, unsigned trackIndex, GridIterator& iterator, Vector<GridItemWithSpan>& itemsSortedByIncreasingSpan, Vector<GridItemWithSpan>& itemsCrossingFlexibleTracks, SingleThreadWeakHashSet<RenderBox>& itemsSet, LayoutUnit currentAccumulatedMbp)
+void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track, unsigned trackIndex, GridIterator& iterator, Vector<GridItemWithSpan>& itemsSortedByIncreasingSpan, Vector<GridItemWithSpan>& itemsCrossingFlexibleTracks, SingleThreadWeakHashSet<RenderBox>& itemsSet, SingleThreadWeakListHashSet<RenderBox>& indefiniteItems, LayoutUnit currentAccumulatedMbp)
 {
-    while (auto* gridItem = iterator.nextGridItem()) {
+    SingleThreadWeakHashSet<RenderBox> visitedIndefiniteMasonryItemsInTrack;
```

Comment:
I'm a little confused as to what this set is doing. I *think* I understand the intention based off of some of the comments below, but I"m not 100% sure (seems like to help force us to consider indefinite items over all of the intrinsically sized tracks). Is there a reason why we can't just consider if the item is contained within `indefiniteItems` since that should be the set of those items? I think this would be a good opportunity to provide a high level explanation of the overall architecture in the commit message to help provide a picture of what this patch is attempting to do.